### PR TITLE
[frontend] Filter correlation graphs on common SCOs and indicators by default (#3227)

### PR DESCRIPTION
**Pull request by Gwendoline-FAVRE-FELIX**
### Proposed changes

* Modification of correlation graph queries to display filtering by indicators and stix-cyber-objects in common.
* Add a button to display all the entities in common.

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/3227

### CODE REVIEW

File: opencti-platform/opencti-front/src/utils/Graph.js [hunk 1]
```diff
@@ -460,29 +459,35 @@ export const buildCorrelationData = (
...
   const filteredNodesIds = computeFilteredNodesIds(
-    thisReportOriginalNodes,
+    thisContainerOriginalNodes,
     filterAdjust.stixCoreObjectsTypes,
     filterAdjust.markedBy,
     filterAdjust.createdBy,
   );
-  const thisReportNodes = thisReportOriginalNodes.map((n) => R.assoc('disabled', filteredNodesIds.includes(n.id), n));
-  const thisReportLinkNodes = R.filter(
+  const thisContainerNodes = thisContainerOriginalNodes.map((n) => R.assoc('disabled', filteredNodesIds.includes(n.id), n));
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/utils/Graph.js]:
```suggestion
  const thisContainerNodes = thisContainerOriginalNodes.map((n) =>{...n, disabled:filteredNodesIds.includes(n.id)}
```

Comment (marieflorescontact) [reply opencti-platform/opencti-front/src/utils/Graph.js]: If possible try to remove Ramda 🙂

File: opencti-platform/opencti-front/src/utils/Graph.js [hunk 2]
```diff
@@ -460,29 +459,35 @@ export const buildCorrelationData = (
...
+  const thisContainerLinkNodes = R.filter(
     (n) => n[key] && n.parent_types && n[key].edges.length > 1,
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/utils/Graph.js:490]:
```suggestion
thisContainerNodes.filter((n) => n[key] && n.parent_types && n[key].edges.length > 1)
```

File: opencti-platform/opencti-front/src/utils/Graph.js [hunk 3]
```diff
@@ -512,14 +517,14 @@ export const buildCorrelationData = (
...
-    relatedReportOriginalNodes,
+  )(thisContainerLinkNodes);
+  const relatedContainerFilteredNodeIds = computeFilteredNodesIds(
+    relatedContainerOriginalNodes,
     filterAdjust.stixCoreObjectsTypes,
     filterAdjust.markedBy,
     filterAdjust.createdBy,
   );
-  const relatedReportNodes = relatedReportOriginalNodes.map((n) => R.assoc('disabled', relatedReportFilteredNodeIds.includes(n.id), n));
+  const relatedContainerNodes = relatedContainerOriginalNodes.map((n) => R.assoc('disabled', relatedContainerFilteredNodeIds.includes(n.id), n));
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/utils/Graph.js]: better remove Ramda

File: opencti-platform/opencti-front/src/utils/Graph.js [hunk 4]
```diff
@@ -547,11 +552,11 @@ export const buildCorrelationData = (
...
               R.map((o) => o.node.id, n[key].edges),
             ),
-      )(relatedReportNodes),
+      )(relatedContainerNodes),
     )),
     R.flatten,
-  )(thisReportLinkNodes);
-  const combinedNodes = R.concat(thisReportLinkNodes, relatedReportNodes);
+  )(thisContainerLinkNodes);
+  const combinedNodes = R.concat(thisContainerLinkNodes, relatedContainerNodes);
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/utils/Graph.js]: Ramda

File: opencti-platform/opencti-front/src/private/components/cases/case_rfts/CaseRftKnowledgeGraphBar.jsx [hunk 5]
```diff
@@ -498,6 +502,25 @@ class CaseRftKnowledgeGraphBar extends Component {
...
                 >
+                  {handleToggleQueryMode && (
+                  <Tooltip
+                    title={
+                      currentQueryMode === 'all-entities'
+                        ? t('Show only correlated observables and indicators')
+                        : t('Show all correlated entities')
+                    }
+                  >
+                    <span>
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/cases/case_rfts/CaseRftKnowledgeGraphBar.jsx]: Do we really need a `<span>` here?

File: opencti-platform/opencti-front/src/private/components/cases/case_rfis/CaseRfiKnowledgeGraphBar.jsx [hunk 6]
```diff
@@ -498,6 +502,25 @@ class CaseRfiKnowledgeGraphBar extends Component {
...
                 >
+                  {handleToggleQueryMode && (
+                  <Tooltip
+                    title={
+                      currentQueryMode === 'all-entities'
+                        ? t('Show only correlated observables and indicators')
+                        : t('Show all correlated entities')
+                    }
+                  >
+                    <span>
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/cases/case_rfis/CaseRfiKnowledgeGraphBar.jsx]: Do we really need the `<span>` here?

File: opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeCorrelation.jsx [hunk 7]
```diff
@@ -245,7 +245,7 @@ class ReportKnowledgeCorrelationComponent extends Component {
       LOCAL_STORAGE_KEY,
     );
     this.zoom = R.propOr(null, 'zoom', params);
-    this.graphObjects = R.map((n) => n.node, props.report.objects.edges);
+    this.allGraphObjects = R.map((n) => n.node, props.report.objects.edges);
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/analyses/reports/ReportKnowledgeCorrelation.jsx]: if you can, remove Ramda

File: opencti-platform/opencti-front/src/private/components/cases/case_incidents/IncidentKnowledgeCorrelation.jsx [hunk 8]
```diff
@@ -389,6 +390,29 @@ class IncidentKnowledgeCorrelationComponent extends Component {
     });
   }
 
+  handleToggleQueryMode() {
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/private/components/cases/case_incidents/IncidentKnowledgeCorrelation.jsx:393]: should we extract a method, as the same code is used 5 times?

**Review by marieflorescontact — COMMENTED** [review]
functionally tested ok 👌

File: opencti-platform/opencti-front/src/private/components/cases/case_incidents/IncidentKnowledgeCorrelation.jsx (hunk 8 above)
Comment (lndrtrbn) [reply opencti-platform/opencti-front/src/private/components/cases/case_incidents/IncidentKnowledgeCorrelation.jsx:393]: Would be nice indeed but we need a bigger refactoring on graphs that will happen later. For now I don't think we want to take time on it, the whole file is a copy paste anyway...

File: opencti-platform/opencti-front/src/private/components/cases/case_rfis/CaseRfiKnowledgeGraphBar.jsx (hunk 6 above)
Comment (lndrtrbn) [reply opencti-platform/opencti-front/src/private/components/cases/case_rfis/CaseRfiKnowledgeGraphBar.jsx]: Same question for other components

**Review by labo-flg — APPROVED** [review]

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**